stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1712,7 +1716,7 @@ reevaluate_if_taskgroup_has_results:;
     if (status.compare_exchange_strong(
         assumedStatus, newStatus.completingPendingReadyWaiting(this).status,
         /*success*/ std::memory_order_release,
-        /*failure*/ std::memory_order_acquire)) {
+        /*failure*/ std::memory_order_relaxed)) {
```

Comment:
I don't think this is correct? I think the acquire here is needed to make sure that we see the write by the child task whose results we are ready to consume.